core/src/test/java/org/apache/calcite/test/ReflectiveSchemaTest.java

```diff
@@ -581,6 +581,12 @@ private void check(ResultSetMetaData metaData, String columnName,
         CalciteAssert.that().withSchema("s", CATCHALL);
     with.query("select \"wrapperLong\" / \"primitiveLong\" as c\n"
         + " from \"s\".\"everyTypes\" where \"primitiveLong\" <> 0")
+        .planContains(
+            "final Long input_value = current.wrapperLong;")
+        .planContains(
+            "final boolean input_isNull = input_value == null;")
+        .planContains(
+            "return input_isNull ? (Long) null : Long.valueOf(input_isNull ? 0L : input_value / Long.valueOf(current.primitiveLong));")
```

Comment:
Technically speaking the second check of `input_isNull` is not required in `Long.valueOf(input_isNull `

Reply:
I guess it was implemented in previous codegen.

Reply:
Yes, it is the final step.
We need to wrap `(isNull, value)` as the Expression.
As there are some cases like` (isNull=true, value=0)`, in which the `value` a primitive's default value, we can not return `value` directly.
Therefore, to avoid lossing nullable, generating:
`isNull? (**)null, box(value)`

Reply:
What I mean is current codegen would likely produce
`return input_isNull ? (Long) null : Long.valueOf(input_value / Long.valueOf(current.primitiveLong));`

How are you going to approach that?

Reply:
Current implementation conduct a step-by-step manner, maybe the non-optimized code is more intuitive.

**public Object current() {
     final org.apache.calcite.test.ReflectiveSchemaTest.EveryType current = (org.apache.calcite.test.ReflectiveSchemaTest.EveryType) inputEnumerator.current();
     final Long input_value = current.wrapperLong;
     final boolean input_isNull = input_value == null;
     final long input_value0 = current.primitiveLong;
     final boolean input_isNull0 = false;
     final long binary_call_value = input_isNull || input_isNull0 ? 0L : input_value / Long.valueOf(input_value0);
     final boolean binary_call_isNull = input_isNull || input_isNull0;
     final Long result = binary_call_isNull ? (Long) null : Long.valueOf(binary_call_value);
     return result;
 }**

The implementation introduces some patterns that need be optimized in BlockBuilder.

Reply:
@vlsi I think we need to optimize this kind of code in BlockBuilder.